mutable_tree.go

```diff
@@ -1055,6 +1051,8 @@ func (tree *MutableTree) saveNewNodes(version int64) error {
 		}
 
 		node._hash(version)
+		newNodes = append(newNodes, node)
```

Comment:
it is for the reverse ordering of `newNode` to store the root at last

Reply:
The thought is that a pre-order traversal produced list of `newNodes` might fail part of the way through writing, leaving the root in the db but not the entire changeset?

Isn't the call the `SaveRoot` what identifies a tree as saved? Otherwise how will an application load a tree at version N after a crash which only saved part of the batch?

Reply:
we assume the node with nonce 1 as the root, `SaveRoot` is only used in case of `nil` or `no updates`

Reply:
Ah ok I see that now, makes sense since nonce is assigned by pre-order traversal.